### PR TITLE
`CI`: run all iOS 17 tests

### PR DESCRIPTION
**Pull request by NachoSoto**
Xcode 15.0.1 (#3311) fixed performance issues (partially), so we should be able to run all tests on iOS 17.

See https://discuss.circleci.com/t/severe-performance-problems-with-xcode-15/49205/65?u=nachosoto

### CODE REVIEW

**Comment by NachoSoto** [comment]
Unfortunately the performance issue is still there when running tests that require a host app :(

File: .circleci/config.yml [hunk 1]
```diff
@@ -424,10 +424,10 @@ jobs:
           name: Run tests
           # StoreKit tests currently get stuck.
           # See https://discuss.circleci.com/t/severe-performance-problems-with-xcode-15/49205/11?u=nachosoto
```
Comment (tonidero) [inline .circleci/config.yml:426]: Can we remove this comment now?

**Review by tonidero — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (NachoSoto) [reply .circleci/config.yml:426]: Oops yeah

File: .circleci/config.yml [hunk 2]
```diff
@@ -958,7 +956,7 @@ workflows:
     when: << pipeline.parameters.generate_snapshots >>
     jobs:
       - run-test-ios-17:
-          xcode_version: '15.0.0'
+          xcode_version: '15.0.1'
```
Comment (NachoSoto) [inline .circleci/config.yml:959]: CircleCI initially shipped the update under `Xcode 15.0.0`, but now it's its own `15.0.1`

**Pull request merged**